flink-baseline/flink-canary/src/main/java/com/ness/flink/canary/pipeline/sources/KafkaConfigsGenerator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2021-2023 Ness Digital Engineering
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ness.flink.canary.pipeline.sources;
+
+import com.ness.flink.canary.pipeline.domain.KafkaConfigs;
+import org.apache.flink.api.java.utils.ParameterTool;
+import org.apache.flink.streaming.api.functions.source.SourceFunction;
+
+public class KafkaConfigsGenerator implements SourceFunction<KafkaConfigs> {
+    private static final long serialVersionUID = 1L;
+//    private volatile boolean running = true;
+    private final ParameterTool params;
+
+    /** Create a bounded KafkaConfigsGenerator with the specified params */
+    public KafkaConfigsGenerator(ParameterTool params) {
+        this.params = params;
+    }
+
+    @Override
+    public void run(SourceContext<KafkaConfigs> ctx) {
+
+        // Generate only one data for now
+        KafkaConfigs kafkaConfigs = KafkaConfigs.builder()
+            .bootStrapServers(params.get("bootstrap.servers", "localhost:9092"))
+            .topic(params.get("topic", "test-topic"))
+            .requestTimeoutMs(params.get("request.timeout.ms", "1000"))
+            .connectionMaxIdleMs(params.get("connections.max.idle.ms", "3000"))
+            .build();
+
+        ctx.collect(kafkaConfigs);
+
+    }
+
+    @Override
+    public void cancel() { }
```

Comment:
Don't see any point to have it. Remove it.

Reply:
cancel method needs to be implemented since it is a part of the parent interface SourceFunction. Instead of removing it, I will implement something in it.

Reply:
You will introduce it only if you implement it. For now just remove it pls.

Reply:
Also please do not mark Conversation as "Resolved" by yourself.
It should be done on my side. 
You just have provide answers on my questions and put comment like: "Done" / "Agreed" etc.
In this case I will be aware that you noticed about my comments/question and handled it.

Reply:
ok, my bad.
 it comes from Interface SourceFunction.